src/python_bindings/fastvdf.cpp

```diff
@@ -61,6 +61,24 @@ PYBIND11_MODULE(chiavdf, m) {
         return is_valid;
     });
 
+    // Checks an N wesolowski proof.
+    m.def("create_discriminant_and_verify_n_wesolowski", [] (const py::bytes& challenge_hash,
+                                   const int discriminant_size_bits,
+                                   const string& x_s,
+                                   const string& proof_blob,
+                                   const uint64_t num_iterations,
+                                   const uint64_t recursion) {
+        std::string challenge_hash_str(challenge_hash);
+        std::vector<uint8_t> challenge_hash_bits = std::vector<uint8_t>(challenge_hash_str.begin(), challenge_hash_str.end());
+        std::string x_s_copy(x_s);
+        std::string proof_blob_copy(proof_blob);
```

Comment:
I think making copies of these arguments is unnecessary. especially of the ones that are `std::string` already, they are clearly copied by the python binding. But I would suggest avoiding those copies too, and take `py::bytes` for `x_s` and `proof_blob` as well. You have to take a `string_view` or `span<char const>` of them before dropping the GIL, but the bytes will stick around. `bytes` is immutable in python, so it's safe. The issue is when you use the "buffer protocol" which may also be `bytearray`, which is mutable and might change once you drop the GIL. But that's not an issue here.

Reply:
i am still unsure how pybind handles these C parameters. Are we sure they are safe even if the GIL is dropped? Given the issues we've had I'd rather just use copies and be safe.